### ISSUE TITLE
Ideas & Discussion

### ISSUE THREAD

**Issue by skmp** [emudev-org/LEL#1]
So,

The idea is to lawyer up and draft a (group of) license(s) that allow knowledge sharing and contributions to projects but restricting/protecting against drama.

One idea is to be similarosh to gpl in structure, but restrict derivatives distribution unless they are in the form of a pr/patch set for the main repo, as a way to avoid forks/drama.

Could be a group of licenses similar to CC or gpl/lgpl with a nice questionnaire wizard to help you choose the license

Thoughts?

**Comment by phire** [comment]
I think it would be useful to have a spectrum of restrictions along the "no-forks" restrictions from strict to less strict. This would 

1. **Strict:** Zero coping of code out of the project is allowed. 
2. **Snippets only:** Copying of small functions and groups of functions is allowed
3. **Non-compete:** Forks are allowed, but only if the branding and UI are completely different. Might also include additional restrictions like not collecting any money (including patreon, in-app ads, app-store), and crediting the original project.

**Comment by phire** [comment]
Some questions to consider, from the top of my head:

### Redistribution: 

 * Do people have the right to redistribute unmodified binaries
 * Are people allowed to bundle the unmodified binary as part of a collection or distro  (aka, freeware)
 * Are people allowed to bundle the unmodified binary with something else
 * Do people (distro, package managers) have the right to build new binaries from the unmodified source and redistribute them?
 * Do people have the right to make minor patches to the source in order to build on a distro/platform and redistribute that binary.
   * If so, do are they required to distribute their patches?
   * Does this right extend far enough to allow ports to new operating systems and devices
 * Is there any restriction on redistribution of old binaries?
 
### The PR exception:
 * What happens if a PR is rejected and closed
   Is the submitter required to delete the patch, or is the patch allowed to continue to exist forever?
 * Is there an exception for people hitting the fork button on github?
   * Are they required to delete it after submitting a fork?
   * Are they allowed to push and display a branch that isn't yet a pull request?
   * What if a branch never gets turned into a pull request?
 * Should PRs get the same retribution rights as the main branch? Or should the rights be very limited
 * Are people allowed to include patches from someone elses unaccepted PR in their own?
 * How do we attach the license to patches? Some sort of vitality? Or do we use some sort of contributor agreement or copyright assignment. 
 * Are people allowed to share patches in places other than the PR (for example, a pastebin or gist)
   * This doesn't have to be an outright ban. 
     There could be an exception for patches for the purpose of development while excluding patches for the purpose of bypassing the license.

### Potential workarounds:

These all need to be blocked somehow.

 * Someone distributes the unmodified binary, but with a loader or patcher that modifies it.
   * Or if redistribution isn't allowed, an automatic downloader script or instructions for the user to manually download the binary
 * Someone provides a tool or set of instructions that enable users to build a modified version of the source. 
 * Someone submits a PR and somehow leverages the exceptions granted to them

**Comment by snaphat** [comment]
Keep in mind most of these things break with the OSI definition of open-source which stipulates redistribution cannot be restricted

**Comment by phire** [comment]
Yeah. This license should state upfront that it doesn't meet the OSI definition, nor does it aim to. 

I think the most restrictive license that can be done while still meeting the [OSI's definition](https://en.wikipedia.org/wiki/The_Open_Source_Definition) is:   
 >  *"No distributing modified source/binaries unless it's a fork. Forks must use a different name"*    

Maybe people want such a license. It could theoretically be in-scope of this project.

**Comment by qeeg** [comment]
I mean, forks should definitely still be allowed I think, since people might have different goals for the project, but don't want to start from scratch.

**Comment by snaphat** [comment]
I think anything that didn't allow forks would relegate itself to obscurity, it would just never take off in reality
